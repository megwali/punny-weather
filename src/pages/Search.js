import { useState } from 'react';
import Dashboard from '../components/Dashboard';
import { Button, SearchContainer, SearchInput } from '../components/styledComponents';


const Search = ({ data, query = '', setQuery }) => {
  const [city, setCity] = useState(query);

  return (
    <>
      <SearchContainer>
        <SearchInput
          onChange={({ target }) => setCity(target.value)}
          placeholder="Search city"
          type="search"
          value={city}
        />
        <Button disabled={!city} onClick={() => setQuery(city)}>Search</Button>
      </SearchContainer>

      <Dashboard data={data} loadingMessage={'Doing science... ðŸ›« ðŸ§ª ðŸ”¬ ðŸ›¬'} />
    </>
  );
};

export default Search;
