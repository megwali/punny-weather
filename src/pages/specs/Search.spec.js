import { create } from 'react-test-renderer';
import Search from '../Search';

jest.mock('../../components/Dashboard', () => 'Dashboard');
jest.mock('../../components/styledComponents', () => ({
  Button: 'Button',
  SearchContainer: 'SearchContainer',
  SearchInput: 'SearchInput'
}));


describe('Search', () => {
  it ('renders', () => {
    const data = {};
    const component = create(<Search data={data} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
