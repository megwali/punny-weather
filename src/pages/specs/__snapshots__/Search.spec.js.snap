// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
Array [
  <SearchContainer>
    <SearchInput
      onChange={[Function]}
      placeholder="Search city"
      type="search"
      value=""
    />
    <Button
      disabled={true}
      onClick={[Function]}
    >
      Search
    </Button>
  </SearchContainer>,
  <Dashboard
    data={Object {}}
    loadingMessage="Doing science... ðŸ›« ðŸ§ª ðŸ”¬ ðŸ›¬"
  />,
]
`;
